### ISSUE TITLE
My node_modules dependencies are not being copied to the .app on Mac OS

### ISSUE THREAD

**Issue by jessegavin** [electron-userland/electron-packager#768]
* [x] I have read the [contribution documentation](https://github.com/electron-userland/electron-packager/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [code of conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project follows, as appropriate.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

**Please describe your issue:**
My node_modules dependencies are not being copied to the `Contents/Resources/app/node_modules` directory when I run `electron-packager .`

**Console output when you run electron-packager with the environment variable `DEBUG=electron-packager`. Please include the stack trace if one exists.**

```
  electron-packager Electron Packager 10.1.0 +0ms
  electron-packager Node v9.2.0 +2ms
  electron-packager Host Operating system: darwin (x64) +0ms
  electron-packager Packager Options: {"_":["."],"all":false,"deref-symlinks":true,"derefSymlinks":true,"download":{"strictSSL":true},"overwrite":false,"prune":true,"quiet":false,"dir":".","protocols":[]} +0ms
  electron-packager Target Platforms: darwin +0ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring application name from name in /Users/me/code/b-tracker/package.json +0ms
  electron-packager Inferring appVersion from version in /Users/me/code/b-tracker/package.json +0ms
  electron-packager Inferring target Electron version from electron in /Users/me/code/b-tracker/package.json +2ms
  electron-packager Application name: my-app +18ms
  electron-packager Target Electron version: 1.7.9 +0ms
  electron-packager Ignored path regular expressions: [ '/node_modules/electron($|/)',
  '/node_modules/electron-prebuilt(-compile)?($|/)',
  '/node_modules/electron-packager($|/)',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$' ] +0ms
  electron-packager Downloading Electron with options {"strictSSL":true,"platform":"darwin","arch":"x64","version":"1.7.9"} +0ms
Packaging app for platform darwin x64 using electron v1.7.9
  electron-packager Creating /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64-template +194ms
  electron-packager Extracting /Users/me/.electron/electron-v1.7.9-darwin-x64.zip to /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64-template +0ms
  electron-packager Initializing app in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64 from /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [ '/Users/me/code/b-tracker/my-app-darwin-x64',
  '/Users/me/code/b-tracker/my-app-linux-ia32',
  '/Users/me/code/b-tracker/my-app-linux-x64',
  '/Users/me/code/b-tracker/my-app-linux-armv7l',
  '/Users/me/code/b-tracker/my-app-linux-arm64',
  '/Users/me/code/b-tracker/my-app-mas-x64',
  '/Users/me/code/b-tracker/my-app-win32-ia32',
  '/Users/me/code/b-tracker/my-app-win32-x64' ] +1s
  electron-packager Pruning modules via: npm prune --production +0ms
  electron-packager Renaming Electron to my-app in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/MacOS +2s
  electron-packager Renaming Electron Helper to my-app Helper in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper EH to my-app Helper EH in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper EH.app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper NP to my-app Helper NP in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper NP.app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper.app to my-app Helper.app in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper EH.app to my-app Helper EH.app in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper NP.app to my-app Helper NP.app in /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Moving /var/folders/fv/2p6drtr94pn_5ltg598j6fzm0000gn/T/electron-packager/darwin-x64/my-app-darwin-x64 to /Users/me/code/b-tracker/my-app-darwin-x64 +1ms
Wrote new app to /Users/me/code/b-tracker/my-app-darwin-x64
```

**What command line arguments are you passing? Alternatively, if you are using the API, what
parameters are you passing to the `packager()` function?**

```
electron-packager .
```

**Please provide either a failing minimal testcase (with a link to the code) or detailed steps to
reproduce your problem. Using [electron-quick-start](https://github.com/electron/electron-quick-start)
is a good starting point.**

```
git clone https://github.com/electron/electron-quick-start.git
cd electron-quick-start
yarn add lodash.capitalize
```

Replace main.js with the following

```
const { app } = require('electron');
const capitalize = require('lodash.capitalize');

app.on('ready', () => {
	console.log(capitalize('electron'));
});
```
Now package the app

```
electron-packager .
```

Now when I look in the node_modules folder in my electron-quick-start.app I see nothing..

```
ls -la ./electron-quick-start-darwin-x64/electron-quick-start.app/Contents/Resources/app/node_modules/

total 8
drwxr-xr-x   3 me  staff  102 Dec  7 20:09 .
drwxr-xr-x  12 me  staff  408 Dec  7 20:09 ..
-rw-r--r--   1 me  staff   64 Dec  7 20:09 .yarn-integrity
```

**Comment by malept** [comment]
I can't reproduce this behavior on Linux. Make sure `lodash.capitalize` is in the `dependencies` in your `package.json`, and if it's there, perhaps try `electron-packager . --package-manager=yarn`?

**Comment by jessegavin** [comment]
Here's my `package.json`

```
{
  "name": "electron-quick-start",
  "version": "1.0.0",
  "description": "A minimal Electron application",
  "main": "main.js",
  "scripts": {
    "start": "electron ."
  },
  "repository": "https://github.com/electron/electron-quick-start",
  "keywords": [
    "Electron",
    "quick",
    "start",
    "tutorial",
    "demo"
  ],
  "author": "GitHub",
  "license": "CC0-1.0",
  "devDependencies": {
    "electron": "~1.7.8"
  },
  "dependencies": {
    "lodash.capitalize": "^4.2.1"
  }
}
```

I tried `electron-packager . --package-manager=yarn`. I get the same result.

**Comment by malept** [comment]
I still can't reproduce this on Linux, after I switched to Node 9.

```shell
$ git clone https://github.com/electron/electron-quick-start
$ cd electron-quick-start
$ yarn add lodash.capitalize
$ git diff
diff --git a/package.json b/package.json
index 4386fec..c99174e 100644
--- a/package.json
+++ b/package.json
@@ -18,5 +18,8 @@
   "license": "CC0-1.0",
   "devDependencies": {
     "electron": "~1.7.8"
+  },
+  "dependencies": {
+    "lodash.capitalize": "^4.2.1"
   }
 }
$ yarn add -D electron-packager
$ $(npm bin)/electron-packager . --package-manager=yarn --platform=darwin
Packaging app for platform darwin x64 using electron v1.7.9
Wrote new app to electron-quick-start/electron-quick-start-darwin-x64
$ ls electron-quick-start-darwin-x64/electron-quick-start.app/Contents/Resources/app/node_modules/ -A
lodash.capitalize  .yarn-integrity
```

**Comment by jessegavin** [comment]
Weird, so it seems like a Mac OS issue?

**Comment by malept** [comment]
Only some macOS platforms. For example, all Packager tests run on Linux, macOS, and Windows. There's at least one test that checks that folders in the packaged `node_modules` exist.

**Comment by MarshallOfSound** [comment]
@jessegavin Can you post the DEBUG output when using `--package-manager=yarn`

**Comment by jessegavin** [comment]
I ran the following and the dependencies were copied over to the node_modules directory as expected!

`DEBUG=electron-packager electron-packager . electron-packager . --package-manager=yarn`

The last time I tried the `--package-manager=yarn` option I did not see the same result.

Thinking about this further, it might be due to the fact that sometimes the tree-structure in vscode (which I was using) is not 100% in sync with files on disk.  I wonder if I would have seen it show up if I had hit the "refresh" button after running it the first time with the `--package-manager=yarn` option.

Sorry, this is definitely a user-error. I didn't know about the `--package-manager=yarn` option before posting this issue.

Thanks for your help.
